transport/conn.go

```diff
@@ -81,7 +81,13 @@ func (c *connWriter) loop() {
 			}
 		}
 		c.metrics.InFlight.Add(uint32(size))
-		c.conn.Flush()
+		err := c.conn.Flush()
+		// bufio.Writer doesn't write if error has been recorded, connection needs to be closed.
+		// We can't use writer.Reset() because it would discard any not flushed data.
+		if err != nil {
+			// TODO: notify connection owner that connection needs to be replaced.
+			return
```

Comment:
I think this seems like pretty reasonable behavior only `writer.Reset()` may discard error on writer but as it deletes any not flushed data it may cause for a thread to hang on `sendRequest()` indefinitely as neither connWriter nor connReader processes its request.

Reply:
Yes, once flush failed connection must be closed.